File: py4j/AdditionApplication.py
from py4j.java_gateway import JavaGateway, GatewayParameters
import ipdb
ipdb.set_trace()

gateway = JavaGateway(gateway_parameters=GatewayParameters(port=25333))
random = gateway.jvm.java.util.Random()
number1 = random.nextInt(10)
number2 = random.nextInt(10)
print(number1, number2)

addition_app = gateway.entry_point
value = addition_app.addition(number1, number2)
print(value)

File: py4j/AdditionApplication.java
import py4j.GatewayServer;

public class AdditionApplication {

    public int addition(int first, int second) {
        return first + second;
    }

    public static void main(String[] args) {
        AdditionApplication app = new AdditionApplication();
        // app is now the gateway.entry_point
        GatewayServer server = new GatewayServer(app, 25333);
        server.start();
        System.out.println("Gateway Server Started");
    }
}
